src/BloomExe/WebLibraryIntegration/BookTransfer.cs

```diff
@@ -745,11 +746,17 @@ private void UploadInternal(string folder, BulkUploadProgressDlg dlg, Applicatio
 				{
 					if (blPublishModel.BookIsAlreadyOnServer)
```

Comment:
Is it the case that if you run the uploader again, it will attempt to reupload all books? That doesn't really seem "restartable".

Reply:
No. It creates a text file of books that it has uploaded and places that in the top directory of the tree being uploaded (BloomBulkUploadLog.txt). Until you delete that file, it will skip any book that is in that file. This "BookIsAlreadyOnServer" check is just in case it was already uploaded from somewhere else and matches the process from the normal single book upload.